### DISCUSSION TITLE
What is this for?

### DISCUSSION THREAD

**Discussion by frederikhors** [gluesql/gluesql#1234] [Q&A]
I just discovered the existence of this project but I don't quite understand what it is for.

What is the difference between this and [sqlx](https://github.com/launchbadge/sqlx) for example?

And are there any benchmarks to understand the performance differences between the various ORMs/query builders?

**Comment by edezhic** [comment]
I'm not a part of this project and I've had the same question initially - as of now I don't think it can compete with sqlx+sqlite combo in benchmarks. But, it's rust-first and compiles into wasm, and with a couple of utils it can provide really great DX. As of now I'm integrating it into my project - https://github.com/edezhic/prest and this is how simple todo app looks like as of now:

```rust
#[derive(Table, Default, serde::Serialize, serde::Deserialize)]
#[serde(default)]
struct Todo {
    #[serde(default = "generate_uuid")]
    pub id: Uuid,
    pub task: String,
    pub done: bool,
}
...
get(|| async { html!(@for todo in Todo::find_all() {(todo)}) })
.put(|Form(todo): Form<Todo>| async move { todo.save().unwrap().render() })
.patch(|Form(mut todo): Form<Todo>| async move { todo.update_done(!todo.done).unwrap().render() })
.delete(|Form(todo): Form<Todo>| async move { todo.remove().unwrap(); }),
```

templates omited, but the fullstack todo app is just 50 LoC, you can find it here - https://github.com/edezhic/prest/blob/main/examples/app-todo/src/main.rs
